src/main/java/frc/robot/subsystems/gpm/Shooter.java

```diff
@@ -0,0 +1,199 @@
+package frc.robot.subsystems.gpm;
+
+import com.revrobotics.CANSparkFlex;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.RelativeEncoder;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.ShooterConstants;
+
+public class Shooter extends SubsystemBase {
+    // constants for Pids
+    public static final double TOP_P = 0.00005;
+    public static final double TOP_I = 0;
+    public static final double TOP_D = 0;
+
+    public static final double S = 0;
+    public static final double V = 1.0/6000;
+
+    public static final double BOTTOM_P = 0.00005;
+    public static final double BOTTOM_I = 0;
+    public static final double BOTTOM_D = 0;
+
+    /**
+     * In RPM
+     */
+    public static final double TOLERANCE = 10;
+
+    private final CANSparkFlex bottomMotor = new CANSparkFlex(ShooterConstants.BOTTOM_MOTOR_ID, MotorType.kBrushless);
+    private final CANSparkFlex topMotor = new CANSparkFlex(ShooterConstants.TOP_MOTOR_ID, MotorType.kBrushless);
+    private final RelativeEncoder bottomMotorEncoder = bottomMotor.getEncoder();
+    private final RelativeEncoder topMotorEncoder = topMotor.getEncoder();
```

Comment:
Is RelativeEncoder the right class? There's a SparkFlexExternalEncoder that implements RelativeEncoder, which might be better.

Reply:
.getEncoder() returns a RelativeEncoder. We are not using an external encoder.